x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1570,12 +1570,12 @@ class AgentPolicyService {
   private checkAgentless(agentPolicy: Partial<NewAgentPolicy>) {
     const cloudSetup = appContextService.getCloud();
     if (
-      (!cloudSetup?.isServerlessEnabled ||
-        !appContextService.getExperimentalFeatures().agentless) &&
-      agentPolicy?.supports_agentless !== undefined
+      (!cloudSetup?.isServerlessEnabled || !cloudSetup?.isCloudEnabled) &&
+      !appContextService.getExperimentalFeatures().agentless &&
+      agentPolicy?.supports_agentless
```

Comment:
would this satisfies all these conditions?


```js
const { isAgentlessEnabled } = useAgentless();

if (!isAgentlessEnabled && agentPolicy?.supports_agentless) {
```

Reply:
This is in the server so hooks are not available,  AFAIK.